docs/specification.md

```diff
@@ -0,0 +1,1280 @@
+
+**getTotalMinted**
+
+Types
+```
+tokenId = nat
+
+getTotalMintedResponse =
+  ( tokenId :tokenId
+  , nat     :totalMinted
+  )
+
+getTotalMintedParam =
+  ( list tokenId                           :tokenIds
+  , contract (list getTotalMintedResponse) :callback
+  )
+
+getTotalMinted = getTotalMintedParam
+```
+
+Parameter (in Michelson)
+```
+(pair %getTotalMinted
+  (list %tokenIds nat)
+  (contract %callback
+    (list
+      (pair %getTotalMintedResponse
+        (nat %tokenId)
+        (nat %totalMinted)
+      )
+    )
+  )
+)
+```
+- Returns current amount of minted coins for an address with a specific token
+  type.
+
+-- TODO: clarify under which conditions and how should
+-- token reset this allowance
+
+**setMintingAllowanceResetInterval** timestamp
+
+Types
+```
+setMintingAllowanceResetInterval =
+  ( tokenId   :tokenId
+  , timestamp :interval
+  )
+
+```
+
+Parameter (in Michelson)
+```
+(pair %setMintingAllowanceResetInterval
+  (address %sender)
+  (timestamp %interval)
+)
+```
+
+- Set the default interval of minting allowance reset.
+
+- Set to 0 to make disable minting allowance reset by stablecoin token.
+
+**mint (TBD)**
+
+Types
+```
+tokenId = nat
+
+mintParam
+  ( address :sender
+  , address :minter
+  , tokenId :tokenId
+  , nat     :value
+  )
+
+mint = list mintParam
+```
+
+Parameter (in Michelson):
+```
+(list %mint
+  (pair
+    (address %sender)
+    (pair
+      (address %minter)
+      (pair
+        (nat %tokenId)
+        (nat %value)
+  )))
+)
+```
+
+- Produces the given amounts of tokens to the wallets associated with the
+  given addresses.
+
+- Each minting must happen atomically, si if one of them fails, then the whole
+  operation must fail.
+
+- The operation must follow permission policies described above.
+
+- Receiving addresses must be whitelisted.
+
+- Sender must be master minter.
+
+- The total amount of minted coins must not exceed the current
+  minting allowance specified for each recepient individually.
+
+- Minting allowance will decrease by the amount of tokens minted
+  and increase the balance of receiver and total minted value.
+
+- Contract must not be paused.
+
+**burn (TBD)**
+
+Types
+```
+tokenId = nat
+
+burnParam
+  ( address :sender
+  , address :minter
+  , tokenId :tokenId
+  , nat     :value
+  )
+
+burn = list burnParam
+```
+
+Parameter (in Michelson):
+```
+(list %burn
+  (pair
+    (address %sender)
+    (pair
+      (address %minter)
+      (pair
+        (nat %tokenId)
+        (nat %value)
+  )))
+)
+```
+
+- Decreases balances for specified wallets and their token types and the total   supply of tokens by the given amount.
+
+- Each minting must happen atomically, si if one of them fails, then the whole
+  operation must fail.
+
+- The operation must follow permission policies described above.
+
+- Sender must be minter and the receiver must have a sufficient amount of
+  funds to be destroyed and be in a sender's minter list.
```

Comment:
AFAIU minter can burn his tokens _only._ Do I get something wrong? The CENTRE token spec says:
>  The minter specifies the amount of tokens to burn, and the minter must have a balance greater than or equal to the amount. Burning tokens is restricted to minter addresses to avoid accidental burning of tokens by end users.